File: src/DesktopApp/CombatAnalysis.Core/Models/GameLogs/DamageDoneGeneralModel.cs
using CombatAnalysis.Core.Interfaces.Entities;

namespace CombatAnalysis.Core.Models.GameLogs;

public class DamageDoneGeneralModel : IGeneralDetailsEntity
{
    public int Id { get; set; }

    public int GameSpellId { get; set; }

    public string Spell { get; set; } = string.Empty;

    public int Value { get; set; }

    public double DamagePerSecond { get; set; }

    public int CritNumber { get; set; }

    public int MissNumber { get; set; }

    public int CastNumber { get; set; }

    public int MinValue { get; set; }

    public int MaxValue { get; set; }

    public double AverageValue { get; set; }

    public bool IsPet { get; set; }

    public int CombatPlayerId { get; set; }
}

File: src/Web/CombatAnalysis.EnhancedWebApp/combatanalysis.enhancedwebapp.client/src/features/gameLogs/types/DamageDoneGeneralModel.ts
export type DamageDoneGeneralModel = {
    id: string;
    spell: string;
    damagePerSecond: number;
    value: number;
    critNumber: number;
    missNumber: number;
    castNumber: number;
    minValue: number;
    maxValue: number;
    averageValue: number;
    isPet: boolean;
    unitId: string;
}